### PR TITLE
Add association image

### PR DESCRIPTION
**Pull request by abocquet**

### CODE REVIEW

File: associations/urls.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
 router.register(r'transactions', TransactionViewSet)
 router.register(r'funding', FundingViewSet)
 urlpatterns += [
+    path("associations/<association_pk>/image", set_association_image),
```
Comment (Piwakk) [inline associations/urls.py]: Is the omission of the trailing `/` intentional?

File: associations/views/association.py [hunk 2]
```diff
@@ -56,7 +64,21 @@ class AssociationViewSet(viewsets.ModelViewSet):
...
+    association = Association.objects.get(pk=association_pk)
+    role = request.user.get_role(association)
+    if not (role and role.administration):
+        raise PermissionDenied()
+
+    serializer = AssociationImageSerializer(data=request)
+    serializer.is_valid(raise_exception=True)
+    serializer.update(association, serializer.validated_data)
+
+    return Response(status=200)
```
Comment (Piwakk) [inline associations/views/association.py]: Every time I test a create function, the default response of DRF is a 201, so I believe it should be the same here

File: associations/urls.py (hunk 1 above)
Comment (abocquet) [reply associations/urls.py]: nope

File: associations/views/association.py (hunk 2 above)
Comment (abocquet) [reply associations/views/association.py]: this is not a creation, but an update. Maybe the code should not be 200, but I beleive it's not 201 either

Comment (Piwakk) [reply associations/views/association.py]: Actually, maybe the http verb to use is `PUT`.
Also see the MDN for the response codes (yeah, there are several cases…) https://developer.mozilla.org/fr/docs/Web/HTTP/M%C3%A9thode/PUT#R%C3%A9ponses

File: associations/urls.py [hunk 3]
```diff
@@ -56,6 +57,7 @@
 router.register(r"transactions", TransactionViewSet)
 router.register(r"funding", FundingViewSet)
 urlpatterns += [
+    path("associations/<association_pk>/image", set_association_logo),
     path("marketplace/balance/", BalanceView.as_view(), name="balance-list"),
```
Comment (Piwakk) [inline associations/urls.py]: No trailing / ?
Sorry, I forgot what you said last time about it

**Pull request merged**